### DISCUSSION TITLE
Possible to use as ts bundler without transpile?

### DISCUSSION THREAD

**Discussion by jeremy-coleman** [timocov/dts-bundle-generator#311] [Q&A] [answered]
Ive been trying to find a good solution for bundling ts files without transpiling, essentially The same thing this does, except code (remaining as ts) not just types. Any thoughts?

**Comment by timocov** [comment]
Hey, did I understand you correctly, that you want to bundle TypeScript code, right? If so, could you please tell why do you need this?)

**Comment by jeremy-coleman** [comment]
yes that is correct. There are many reasons to list, but generally how typescript files are resolved and the scope of the members within the code differ based on the bundler AND parser, leading to many issues. Using tsc to combine them into a valid program is good for long-term code usability.

**Comment by timocov** [comment, marked as answer]
I didn't face a tool for that (actually I event don't know why I need to use it at all). But as an idea you can start from rollup and what it does (for example, afaik https://github.com/Swatinem/rollup-plugin-dts uses that approach to bundle dts).
